clinica/pipelines/statistics_surface/_model.py

```diff
@@ -0,0 +1,728 @@
+
+
+class GLMFactory:
+    """Factory class for building GLM models.
+
+    Attributes
+    ----------
+    feature_label: Label used for building output filenames.
+    """
+
+    def __init__(self, feature_label: str) -> None:
+        self.feature_label = feature_label
+
+    def create_model(
+        self,
+        glm_type: str,
+        design: str,
+        df: pd.DataFrame,
+        contrast: str,
+        **kwargs,
+    ) -> GLM:
+        """Factory method for building a GLM model instance corresponding to the
+        provided type and design matrix.
+
+        Parameters
+        ----------
+        glm_type: Type of GLM to be created. Either "correlation" or "group_comparison".
+        design: Design matrix in string format. If this contains a "*", it will be
+            interpreted as an interaction effect.
+        df: Subjects DataFrame.
+        contrast: Contrast in string format.
+        """
+        cprint(
+            msg=f"The GLM model is: {design} and the GLM type is: {glm_type}",
+            lvl="info",
+        )
+        if glm_type == "correlation":
+            return CorrelationGLM(design, df, self.feature_label, contrast, **kwargs)
+        elif glm_type == "group_comparison":
+            if "*" in design:
+                warnings.warn(
+                    "You included interaction as covariate in your model, "
+                    "please carefully check the format of your tsv files."
+                )
+                return GroupGLMWithInteraction(
+                    design, df, self.feature_label, contrast, **kwargs
+                )
+            return GroupGLM(design, df, self.feature_label, contrast, **kwargs)
+        raise ValueError(
+            f"GLM factory received an unknown GLM type: {glm_type}."
+            f"Only 'correlation' and 'group_comparison' are supported."
+        )
+
+
+class PValueResults:
+    """This class implements a container for raw (uncorrected)
+    P-value results obtained with a GLM model.
+
+    Attributes
+    ----------
+    pvalues: Array of uncorrected P-values.
+    mask: Binary mask.
+    thresh: Threshold.
+    """
+
+    def __init__(
+        self,
+        pvalues: np.ndarray,
+        mask: np.ndarray,
+        threshold: float,
+    ) -> None:
+        self.P = pvalues
+        self.mask = mask
+        self.thresh = threshold
+
+    @classmethod
+    def from_t_statistics(
+        cls,
+        tstats: np.ndarray,
+        df: pd.DataFrame,
+        mask: np.ndarray,
+        threshold: float,
+    ):
+        """Instantiate the class from an array of T-statistics.
+
+        Parameters
+        ----------
+        tstats: Array of T-statistics.
+        df: Subjects DataFrame.
+        mask: Binary mask.
+        threshold: Threshold.
+        """
+        from scipy.stats import t
+
+        return cls(1 - t.cdf(tstats, df), mask, threshold)
+
+    def to_dict(self, jsonable: bool = True):
```

Comment:
I am not sure what `jsonsable` provides. Maybe you want both a `to_dict` and a `to_json` method?

Reply:
I agree with this comment and the one below about duplicated `to_dict` method.
I made some refactoring in [3931d6e](https://github.com/aramis-lab/clinica/pull/669/commits/3931d6e5684151e63d8dffc396da84f1e71495b6). Let me know what you think.